Architecture.md

```diff
@@ -12,7 +12,33 @@
 
 #### Web application
 
-***TODO: [CLIENT] fill for web application frontend***
+The website is made using [React](https://react.dev/), we decided to use this framework because it is the most popular, widely used in industry and greatly documented. Furthermore, the whole team was already familiar with React thanks to several past projects.
+
+As for the component library, we chose [Chakra UI](https://chakra-ui.com/). Even though we didn't have any experience with it, we decided to try it out since it was visualy appealing and popular. It is also very well documented and supported by the community.
+
+[Typescript](https://www.typescriptlang.org/) was a must have for us to ensure code quality and to avoid bugs. Using a strongly typed codebase is also a good way to ensure the maintainability of our project in the future.
+
+The website sends requests to the client API with the use of Axios HTTP Client. Axios has several advantages as compared to the more well-known Fetch API. It [supports more browsers](https://github.com/axios/axios#browser-support), has built-in protection against CSRF (Cross-Site Request Forgery) and converts data to json format automatically. 
+All of this makes Axios one of the most straightforward solutions to use.
+To simplify the development of the connection with API, our team have used the [OpenAPI Generator](https://openapi-generator.tech/) tool which generated typescript objects and functions for every controller and endpoint in our API. The autogenerated code uses Axios to send requests and is based on the OpenAPI specification of the API which was conveniently provided to us by NestJS once we added the `SwaggerModule`.
+
+**Website structure**
+The web application offers to users several public and protected routes, which give access to the main PaaSTech features. 
```

Comment:
```suggestion
The web application offers to users several public and protected routes, which gives access to the main PaaSTech features. 
```

Reply:
Here "which" refers to routes and not the application, so it is correct to use "give" instead of "gives".